viper.go

```diff
@@ -313,6 +313,8 @@ func (v *Viper) searchMap(source map[string]interface{}, path []string) interfac
 
 	if next, ok := source[path[0]]; ok {
 		switch next.(type) {
+		case map[interface{}]interface{}:
```

Comment:
I believe another way in which you get a map with `interface{}` as a key is when a YAML config is parsed in. In any case looks good, will try to get it merged tomorrow.